community/kernel/src/main/java/org/neo4j/kernel/impl/locking/community/CommunityLockClient.java

```diff
@@ -44,129 +61,243 @@ public CommunityLockClient( LockManagerImpl manager )
     @Override
     public void acquireShared( Locks.ResourceType resourceType, long resourceId )
     {
-        PrimitiveLongObjectMap<LockResource> localLocks = localShared( resourceType );
-        LockResource resource = localLocks.get( resourceId );
-        if( resource != null )
+        if ( !stateHolder.incrementActiveClients() )
         {
-            resource.acquireReference();
+            return;
```

Comment:
Agree, in practice for __now__ that should never be the case but still that's wrong. Will throw some sort of already closed exception.